trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -168,17 +190,17 @@ public function insertTrait($trait, $schema = NULL) {
       ]
     ];
 
-    // Create trait.
+    // Create trait: Trait, Method and Unit.
     foreach($arr_trait as $type => $values) {
-      // Check if a record exists in the same Genus.
-      $value = ($type == 'method') ? $values['description'] : $values['name'];
-      $id = $this->chado->query($sql, [':value' => $value, ':id' => $genus_config[ $type ]['id']])
-        ->fetchField();
+      // Inspect cvterm to see if the trait asset already existed.
+      $trait_asset_rec = $this->getTraitAsset($values['name'], $type);
```

Comment:
As mentioned previously I find this naming very confusing and not very informative. I think it would greatly improve readability and maintainability to refer to the fact that this is returning a cvterm.
```suggestion
      $trait_cvterm = $this->getCvterm($values['name'], $type);
```